### PR TITLE
Retrieve CSRF token before each mutation

### PR DESCRIPTION
**Pull request by mtfranchetto**
## Description

Align the frontend after the backend changes for the new CSRF token generation mechanism #495 

## Changes
- add the ability to pass headers to the `executeRequest` function
- moved the CSRF token retrieval from the initial state to the `executeRequest` function
- stub `executeRequest` instead of `axios` in `CreateCluster` tests, otherwise the CSRF check complains

## How Has This Been Tested?

Run the portal locally and verified:
- no CSRF call is made at the startup
- a CSRF token is retrieved before each mutation

## PR Quality Checklist

- [x] I added tests to new or existing code
- [ ] I removed hardcoded strings and used our `i18n` solution instead (see [here](https://github.com/aws-samples/pcluster-manager/pull/175/commits/fdc6b77987c87a26f51dbc8da5d371d95ef80601))
- [ ] I made sure no sensitive info gets logged at any time in the codebase (see [here](https://cheatsheetseries.owasp.org/cheatsheets/Logging_Cheat_Sheet.html)) (e.g. no user info or details, no stacktraces, etc.)
- [ ] I checked that infrastructure/update_infrastructure.sh runs without any error
- [x] I checked that `npm run build` builds without any error
- [x] I checked that clusters are listed correctly
- [ ] I checked that a new cluster can be created (config is produced and dry run passes)
- [x] I checked that login and logout work as expected

In order to increase the likelihood of your contribution being accepted, please make sure you have read both the [Contributing Guidelines](../CONTRIBUTING.md) and the [Project Guidelines](../PROJECT_GUIDELINES.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: frontend/src/model.tsx [hunk 1]
```diff
@@ -899,24 +895,10 @@ async function GetAppConfig() {
...
 
-async function LoadCsrfToken(logger: ILogger): Promise<string | null> {
-  try {
-    const token = await getCsrfToken(axiosInstance)
-    setCsrfTokenHeader(axiosInstance, token)
-    logger.info(`Set X-CSRF-Token header to ${token}`)
-    return token
-  } catch (error) {
-    logger.error('Could not fetch CSRF token')
-    notify(i18n.t('global.errors.csrfError'), 'error')
```
Comment (mendaomn) [inline frontend/src/model.tsx:910]: nit-pick: we can get rid of `global.errors.csrfError`

**Review by BarcoMasile — DISMISSED** [review]

File: frontend/src/http/__tests__/csrf.test.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          body: 'test',
+        })
+
+        expect(data).toMatchObject({
+          success: true,
+        })
+      })
+    })
+
+    describe('when a CSRF token request fails', () => {
```
Comment (mendaomn) [inline frontend/src/http/__tests__/csrf.test.ts:59]: question: double checking I got this right:
- the caller will handle the error
- logging is delegated to the axios interceptors

is that correct?

Comment (mtfranchetto) [reply frontend/src/http/__tests__/csrf.test.ts:59]: Correct!

File: frontend/src/model.tsx (hunk 1 above)
Comment (mtfranchetto) [reply frontend/src/model.tsx:910]: Right!

Comment (mtfranchetto) [reply frontend/src/model.tsx:910]: Removed

**Review by mendaomn — APPROVED** [review]

**Pull request merged**